### DISCUSSION TITLE
Support for Non-English Numeric Input in NumberInput Component

### DISCUSSION THREAD

**Discussion by mnakhli** [mantinedev/mantine#6899] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

Latest

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

The current `NumberInput` component only accepts English (Latin) numerals, restricting input from users who use other numeral systems such as Arabic, Persian, or others. This limitation affects users with keyboards or language settings that default to these numerals.

**Expected Behavior:**
- The component should either:
   1. Accept numerals from other languages (e.g., Arabic or Persian) directly as valid input.
   2. Automatically convert non-English numerals to their English counterparts before processing.

**Actual Behavior:**
- Only English numerals are allowed, and any attempt to input numerals from other numeral systems is blocked.

**Suggested Solution:**
- Implement a method to detect and transform numerals from other languages (like Arabic or Persian) into English numerals upon input. This will ensure that all users, regardless of their keyboard settings, can enter valid numeric data.

**Steps to Reproduce:**
1. Set your keyboard to a layout that uses Arabic or Persian numerals.
2. Try to enter a number into the NumberInput component.
3. Notice that the input is rejected unless the keyboard is switched back to English numerals.

**Additional Context:**
Supporting multiple numeral systems would enhance accessibility and usability for international users, improving the inclusivity of the application.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Currently, I do not plan to implement this feature in the near future
